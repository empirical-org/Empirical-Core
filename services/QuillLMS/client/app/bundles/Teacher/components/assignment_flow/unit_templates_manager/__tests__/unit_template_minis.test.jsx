import React from 'react';
import { shallow } from 'enzyme';

import UnitTemplateMinis from '../unit_template_minis';

const sharedProps = {
  types: [{name: "Diagnostic", id: "diagnostic"}],
  displayedModels: {
    20: {
      id: 20,
      name: "Sentence Structure Diagnostic",
      time: 30,
      order_number: 0,
      created_at: 0,
      number_of_standards: 1,
      activity_info: "test activity",
      author: {
        name: "author",
        avatar_url: "avatar"
      },
      unit_template_category: {
        primary_color: "#00c2a2",
        secondary_color: "#027360",
        name: "Diagnostic",
        id: 9
      },
      activities: [],
      type: {
        name: "Diagnostic",
        primary_color: "#ea9a1a"
      },
      grades: ['1','2']
    }
  },
  data: {
    categories: [
      { primary_color: "#00c2a2",
        secondary_color: "#027360",
        name: "Diagnostic",
        id: 9}
    ],
    stage: "index",
    displayedModels: {
      20: {
        id: 20,
        name: "Sentence Structure Diagnostic",
        time: 30,
        order_number: 0,
        created_at: 0,
        number_of_standards: 1,
        activity_info: "test activity",
        author: {
          name: "author",
          avatar_url: "avatar"
        },
        unit_template_category: {
          primary_color: "#00c2a2",
          secondary_color: "#027360",
          name: "Diagnostic",
          id: 9
        },
        activities: [],
        type: {
          name: "Diagnostic",
          primary_color: "#ea9a1a"
        },
        grades: ['1','2']
      }
    }
  }
}

describe('UnitTemplateMinis component', () => {

  it('should render without createYourOwn mini when teacher not signed in', () => {
    const wrapper = shallow(<UnitTemplateMinis {...sharedProps} signedInTeacher={false} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('UnitTemplateMini').findWhere(n => n.prop('data').id === 'createYourOwn')).toHaveLength(0)
  });

  it('should show createYourOwn mini when teacher is signed in', () => {
    const wrapper = shallow(<UnitTemplateMinis {...sharedProps} signedInTeacher={true} />)
    expect(wrapper.find('UnitTemplateMini').findWhere(n => n.prop('data').id === 'createYourOwn')).toHaveLength(1)
  })
  it('should render dropdowns when on mobile', () => {
    const wrapper = shallow(<UnitTemplateMinis {...sharedProps} signedInTeacher={true} />)
    wrapper.setState({ onMobile: true });
    expect(wrapper).toMatchSnapshot();
  })

});
