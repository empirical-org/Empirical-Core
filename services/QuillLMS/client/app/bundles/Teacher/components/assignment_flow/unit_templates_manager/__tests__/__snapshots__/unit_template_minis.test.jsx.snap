// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnitTemplateMinis component should render dropdowns when on mobile 1`] = `
<div
  className="unit-template-minis-container"
  key="always-display"
>
  <AssignmentFlowNavigation />
  <div
    className="container"
  >
    <div>
      <h1>
        Select an activity pack for your students.
      </h1>
      <div>
        <div
          className="filter-options"
        >
          <div
            className="dropdowns"
          >
            <DropdownInput
              className="grade-level-dropdown"
              label="Grade level range"
              options={
                Array [
                  Object {
                    "label": "All",
                    "link": "/assign/featured-activity-packs",
                    "value": null,
                  },
                  Object {
                    "label": "4th-12th",
                    "link": "/assign/featured-activity-packs?gradeLevel=4th-12th",
                    "value": "4th-12th",
                  },
                  Object {
                    "label": "6th-12th",
                    "link": "/assign/featured-activity-packs?gradeLevel=6th-12th",
                    "value": "6th-12th",
                  },
                  Object {
                    "label": "8th-12th",
                    "link": "/assign/featured-activity-packs?gradeLevel=8th-12th",
                    "value": "8th-12th",
                  },
                  Object {
                    "label": "10th-12th",
                    "link": "/assign/featured-activity-packs?gradeLevel=10th-12th",
                    "value": "10th-12th",
                  },
                ]
              }
              value={
                Object {
                  "label": "All",
                  "link": "/assign/featured-activity-packs",
                  "value": null,
                }
              }
            />
            <DropdownInput
              className="category-dropdown"
              label="Pack type"
              options={
                Array [
                  Object {
                    "label": "All",
                    "link": "/assign/featured-activity-packs",
                    "value": null,
                  },
                  Object {
                    "label": "Diagnostic",
                    "link": "/assign/featured-activity-packs?category=Diagnostic",
                    "value": 9,
                  },
                ]
              }
              value={
                Object {
                  "label": "All",
                  "link": "/assign/featured-activity-packs",
                  "value": null,
                }
              }
            />
            <DropdownInput
              className="category-dropdown"
              handleChange={[Function]}
              label="View"
              options={
                Array [
                  Object {
                    "label": "Grid",
                    "value": "grid",
                  },
                  Object {
                    "label": "List",
                    "value": "list",
                  },
                ]
              }
              value={
                Object {
                  "label": "Grid",
                  "value": "grid",
                }
              }
            />
          </div>
          <DropdownInput
            className="pack-type-dropdown"
            handleChange={[Function]}
            options={
              Array [
                Object {
                  "label": "All Packs",
                  "value": "/activities/packs",
                },
                Object {
                  "label": "Diagnostic",
                  "value": "/assign/featured-activity-packs?type=diagnostic",
                },
              ]
            }
            value={
              Object {
                "label": "All Packs",
                "value": "/activities/packs",
              }
            }
          />
        </div>
        <div
          className="unit-template-minis"
        >
          <section
            className="all-packs-section"
          >
            <p
              className="pack-type-header"
            >
              Reading Texts
            </p>
            <section
              className="packs-section"
            />
          </section>
          <section
            className="all-packs-section"
          >
            <p
              className="pack-type-header"
            >
              Diagnostic
            </p>
            <section
              className="packs-section"
            >
              <UnitTemplateMini
                data={
                  Object {
                    "activities": Array [],
                    "activity_info": "test activity",
                    "author": Object {
                      "avatar_url": "avatar",
                      "name": "author",
                    },
                    "created_at": 0,
                    "grades": Array [
                      "1",
                      "2",
                    ],
                    "id": 20,
                    "name": "Sentence Structure Diagnostic",
                    "number_of_standards": 1,
                    "order_number": 0,
                    "time": 30,
                    "type": Object {
                      "name": "Diagnostic",
                      "primary_color": "#ea9a1a",
                    },
                    "unit_template_category": Object {
                      "id": 9,
                      "name": "Diagnostic",
                      "primary_color": "#00c2a2",
                      "secondary_color": "#027360",
                    },
                  }
                }
                index={0}
                key="20"
                signedInTeacher={true}
              />
            </section>
          </section>
          <section
            className="all-packs-section"
          >
            <p
              className="pack-type-header"
            >
              Language Skills
            </p>
            <section
              className="packs-section"
            />
          </section>
          <section
            className="all-packs-section"
          >
            <p
              className="pack-type-header"
            >
              Whole Class Lessons
            </p>
            <section
              className="packs-section"
            >
              <UnitTemplateMini
                data={
                  Object {
                    "id": "createYourOwn",
                    "non_authenticated": undefined,
                    "unit_template_category": null,
                  }
                }
                index={0}
                key="createYourOwn"
                signedInTeacher={true}
              />
            </section>
          </section>
        </div>
        <div />
      </div>
    </div>
  </div>
</div>
`;

exports[`UnitTemplateMinis component should render without createYourOwn mini when teacher not signed in 1`] = `
<div
  className="unit-template-minis-container"
  key="always-display"
>
  <div
    className="container"
  >
    <div>
      <h1>
        Select an activity pack for your students.
      </h1>
      <div>
        <div
          className="filter-options"
        >
          <div
            className="dropdowns"
          >
            <DropdownInput
              className="grade-level-dropdown"
              label="Grade level range"
              options={
                Array [
                  Object {
                    "label": "All",
                    "link": "/activities/packs",
                    "value": null,
                  },
                  Object {
                    "label": "4th-12th",
                    "link": "/activities/packs?gradeLevel=4th-12th",
                    "value": "4th-12th",
                  },
                  Object {
                    "label": "6th-12th",
                    "link": "/activities/packs?gradeLevel=6th-12th",
                    "value": "6th-12th",
                  },
                  Object {
                    "label": "8th-12th",
                    "link": "/activities/packs?gradeLevel=8th-12th",
                    "value": "8th-12th",
                  },
                  Object {
                    "label": "10th-12th",
                    "link": "/activities/packs?gradeLevel=10th-12th",
                    "value": "10th-12th",
                  },
                ]
              }
              value={
                Object {
                  "label": "All",
                  "link": "/activities/packs",
                  "value": null,
                }
              }
            />
            <DropdownInput
              className="category-dropdown"
              label="Pack type"
              options={
                Array [
                  Object {
                    "label": "All",
                    "link": "/activities/packs",
                    "value": null,
                  },
                  Object {
                    "label": "Diagnostic",
                    "link": "/activities/packs?category=Diagnostic",
                    "value": 9,
                  },
                ]
              }
              value={
                Object {
                  "label": "All",
                  "link": "/activities/packs",
                  "value": null,
                }
              }
            />
            <DropdownInput
              className="category-dropdown"
              handleChange={[Function]}
              label="View"
              options={
                Array [
                  Object {
                    "label": "Grid",
                    "value": "grid",
                  },
                  Object {
                    "label": "List",
                    "value": "list",
                  },
                ]
              }
              value={
                Object {
                  "label": "Grid",
                  "value": "grid",
                }
              }
            />
          </div>
          <div
            className="type-options"
          >
            <Link
              className="focus-on-light active"
              to="/activities/packs"
            >
              All Packs
            </Link>
            <Link
              className="focus-on-light "
              key="diagnostic"
              to="/activities/packs?type=diagnostic"
            >
              Diagnostic
            </Link>
          </div>
        </div>
        <div
          className="unit-template-minis"
        >
          <section
            className="all-packs-section"
          >
            <p
              className="pack-type-header"
            >
              Reading Texts
            </p>
            <section
              className="packs-section"
            />
          </section>
          <section
            className="all-packs-section"
          >
            <p
              className="pack-type-header"
            >
              Diagnostic
            </p>
            <section
              className="packs-section"
            >
              <UnitTemplateMini
                data={
                  Object {
                    "activities": Array [],
                    "activity_info": "test activity",
                    "author": Object {
                      "avatar_url": "avatar",
                      "name": "author",
                    },
                    "created_at": 0,
                    "grades": Array [
                      "1",
                      "2",
                    ],
                    "id": 20,
                    "name": "Sentence Structure Diagnostic",
                    "number_of_standards": 1,
                    "order_number": 0,
                    "time": 30,
                    "type": Object {
                      "name": "Diagnostic",
                      "primary_color": "#ea9a1a",
                    },
                    "unit_template_category": Object {
                      "id": 9,
                      "name": "Diagnostic",
                      "primary_color": "#00c2a2",
                      "secondary_color": "#027360",
                    },
                  }
                }
                index={0}
                key="20"
                signedInTeacher={false}
              />
            </section>
          </section>
          <section
            className="all-packs-section"
          >
            <p
              className="pack-type-header"
            >
              Language Skills
            </p>
            <section
              className="packs-section"
            />
          </section>
          <section
            className="all-packs-section"
          >
            <p
              className="pack-type-header"
            >
              Whole Class Lessons
            </p>
            <section
              className="packs-section"
            />
          </section>
        </div>
        <div />
      </div>
    </div>
  </div>
</div>
`;
